policy-test/tests/outbound_api_gateway.rs

```diff
@@ -34,250 +34,103 @@ async fn service_does_not_exist() {
 }
 
 #[tokio::test(flavor = "current_thread")]
-async fn egress_switches_to_fallback() {
+async fn service_with_no_http_routes() {
     with_temp_ns(|client, ns| async move {
-        let egress_net = create_egress_network(&client, &ns, "egress-net").await;
-
-        let mut policy_api = grpc::OutboundPolicyClient::port_forwarded(&client).await;
-        let mut rsp = policy_api.watch_ip(&ns, "1.1.1.1", 80).await.unwrap();
-
-        let policy = rsp.next().await.unwrap().unwrap();
-        let meta = policy.metadata.unwrap();
-
-        let expected_meta = meta::Metadata {
-            kind: Some(meta::metadata::Kind::Resource(meta::Resource {
-                group: "policy.linkerd.io".to_string(),
-                port: 80,
-                kind: "EgressNetwork".to_string(),
-                name: "egress-net".to_string(),
-                namespace: ns.clone(),
-                section: "".to_string(),
-            })),
-        };
-
-        assert_eq!(meta, expected_meta);
-
-        delete(&client, egress_net).await;
-        assert!(rsp.next().await.is_none());
-
-        let mut rsp = policy_api.watch_ip(&ns, "1.1.1.1", 80).await.unwrap();
-
-        let policy = rsp.next().await.unwrap().unwrap();
-        let meta = policy.metadata.unwrap();
-        let expected_meta = meta::Metadata {
-            kind: Some(meta::metadata::Kind::Default("egress-fallback".to_string())),
-        };
-        assert_eq!(meta, expected_meta);
+        // Create a service
+        let svc = create_service(&client, &ns, "my-svc", 4191).await;
+        parent_with_no_http_routes(Resource::Service(svc), &client, &ns).await;
     })
     .await;
 }
 
 #[tokio::test(flavor = "current_thread")]
-async fn fallback_switches_to_egress() {
+async fn egress_net_with_no_http_routes() {
```

Comment:
it was a bit surprising to see egress_net tests here when there's a dedicated egress_networks test file.  but I understand why it is so: we need to replicated these tests for every route type (gateway HTTP, linkerd HTTP, and GRPC).  essentially we want to test every combination of the matrix of resource x route.  i.e. (service, egress_network) x (gateway HTTP, linkerd HTTP, GRPC), which means that all of our tests are replicated 6 times.

Ideally we should spend some time figuring out how to test these combinations in a way that doesn't require duplicating each test 6 times.  But that's perhaps outside the score of this PR.

Reply:
Yes I think we can try and abstract over the resource route type, etc but yes this is probably for a follow-up cleanup. I also do not like the duplication.